File: scripts/libraries/project.rb
class Project
	def self.get_repository_directory_path()
		File.dirname(File.dirname(__dir__))
	end

	def self.get_source_images_directory_path()
		File.join(self.get_repository_directory_path(), "source_images")
	end

	def self.get_source_files_directory_path()
		File.join(self.get_repository_directory_path(), "sources")
	end

	def self.get_distribution_directory_path()
		File.join(self.get_repository_directory_path(), "dragon_byte")
	end
end

File: scripts/libraries/project.js
// @ts-check

import path from "path"
import { Directory } from "./file_system.js"

export class Project
{
	/** @returns {Directory} */
	static getRepositoryDirectory()
	{ return (new Directory(path.dirname(path.dirname(path.dirname(import.meta.url))).split("file://")[1])) }

	/** @returns {Directory} */
	static getSourceFilesDirectory()
	{
		return (new Directory(path.join(
			this.getRepositoryDirectory().getPath(),
			"sources"
		)))
	}

	/** @returns {Directory} */
	static getSourceImagesDirectory()
	{
		return (new Directory(path.join(
			this.getRepositoryDirectory().getPath(),
			"source_images"
		)))
	}

	/** @returns {Directory} */
	static getDistributionDirectory()
	{
		return (new Directory(path.join(
			this.getRepositoryDirectory().getPath(),
			"dragon_byte"
		)))
	}
}

